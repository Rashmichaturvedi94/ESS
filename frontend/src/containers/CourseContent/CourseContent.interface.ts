export interface CourseContentProps {
  myProp?: boolean;
  id: number;
}
