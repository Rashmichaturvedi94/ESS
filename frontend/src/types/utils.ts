export type Nullable<T> = {
  [P in keyof T]: T[P] | null;
};

export type Uri = string;

export type DateString = string;

export type LocalOrRemoteFile = File | Uri;

export type Modify<T, R> = Omit<T, keyof R> & R;
