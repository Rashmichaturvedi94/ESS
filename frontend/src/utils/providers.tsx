import React, { ComponentType, FC } from 'react';
import * as Sentry from '@sentry/react';
import { QueryClientProvider } from 'react-query';
import DateFnsUtils from '@date-io/date-fns';
import { ThemeProvider } from 'styled-components';
import { StylesProvider, MuiThemeProvider } from '@material-ui/core';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { queryClient } from '../api';
import { theme } from '../styles/theme';


const createProviderHOC = (Provider: ComponentType) => <P extends object>(
  Component: ComponentType<P>,
) => (props: P) => (
  <Provider>
    <Component {...props} />
  </Provider>
);
const FallbackComponent = () => <div>An error has occurred</div>;

export const Providers: FC = ({ children }) => (
  <Sentry.ErrorBoundary fallback={FallbackComponent} showDialog>
    <StylesProvider injectFirst>
      <MuiThemeProvider theme={theme}>
        <ThemeProvider theme={theme}>
          <MuiPickersUtilsProvider utils={DateFnsUtils}>
            <QueryClientProvider client={queryClient}>
              {children}
            </QueryClientProvider>
          </MuiPickersUtilsProvider>
        </ThemeProvider>
      </MuiThemeProvider>
    </StylesProvider>
  </Sentry.ErrorBoundary>
);

export const withProviders = createProviderHOC(Providers);
