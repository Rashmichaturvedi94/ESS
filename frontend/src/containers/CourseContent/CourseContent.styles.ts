import { Typography } from '@material-ui/core';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { CourseContentProps } from './CourseContent.interface';

export const CourseContent = styled.div<Pick<CourseContentProps, 'myProp'>>`
flex: 1;
display: flex;
flex-direction: row;
text-decoration: none;
`;

export const GridListTileLink = styled(Link)`
 text-decoration: none;
 color: initial;
 &:focus, &:hover, &:visited, &:link, &:active {
    text-decoration: none;
    color: initial;
}
 &.MuiGridListTile-tile {
 display: flex;
 text-decoration: none;
 color: initial;
 }
`;

export const LessonText = styled(Typography)`
  font-size: 40px;
  font-weight: 700;
`;