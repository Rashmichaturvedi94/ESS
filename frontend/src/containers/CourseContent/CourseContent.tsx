import ReactPlayer from 'react-player';
import React, { FC, useState } from 'react';
import { BrowserRouter as Router, useParams } from 'react-router-dom';
import { Box, GridList, Typography, GridListTile, Divider, Button } from '@material-ui/core';
import { CourseContentProps } from './CourseContent.interface';
import { useCourse } from '../../api';
import { CourseParams, CourseContents } from '../../api/api.interface';
import {
  CourseContent as CourseContentComponent, LessonText
} from './CourseContent.styles';

export const getImage = (courseContent: CourseContents) => typeof courseContent.img === 'string' ? courseContent.img : 'https://content.techgig.com/thumb/msid-79844104,width-860,resizemode-4/5-Best-programming-languages-to-learn-in-2021.jpg?140622';
  
export const CourseContent: FC<CourseContentProps> = () => {

  const { courseId } = useParams<CourseParams>();
  const course = useCourse({ courseId });
  const [currentVideo, setCurrentVideo] = useState('');

  return (
    <CourseContentComponent>
      <Router>
        <React.Fragment key="key">
            <Box flex={1} display="flex" alignItems="center" justifyContent="center">
            <Box style={{ margin: 2}} width="50%" height="100%" justifyContent="center">
              <Box style ={{padding: 2}} justifyContent="center"  alignItems="center">
                <Typography variant="h3" component="h2" align='left'>
                  {course.data?.title}
                </Typography> 
            </Box>
            <ReactPlayer url={currentVideo} />
            </Box>
            <Divider orientation="vertical" style={{color: 'black', width: 5}}/>
            <Box display="flex" flexDirection="column" width={600} gridRowGap={10} paddingLeft = '25px'>
            <GridList>
              <LessonText>Lessons</LessonText>
                {course.data?.contents?.map((content) => (
                  <GridListTile key={content.id} style={{ height: 110, width: 560, border: '2px solid black', margin: 2 }}>
                    <Button onClick={() => setCurrentVideo(content.url ?? '')}>
                      <Box flex={1} display="flex" alignItems="left" justifyContent="top" style={{ backgroundColor: 'white' }}> 
                        <Box>
                          <img src={getImage(content)} style={{ height: 110, width:110, overflow: 'hidden' }} alt={content.title}/>
                        </Box>
                        <Box justifyContent="center" alignItems="left" style={{margin: 2}}>
                          <Typography variant="h5" component="h2" align='left'>{content.title}</Typography>
                          <Typography variant="h6" component="h2" align='left'>{content.description}</Typography>
                        </Box>
                      </Box>
                    </Button>
                  </GridListTile>
                ))}
              </GridList>
              </Box>
              </Box>
          </React.Fragment>
        </Router>
      </CourseContentComponent>
  );
};
