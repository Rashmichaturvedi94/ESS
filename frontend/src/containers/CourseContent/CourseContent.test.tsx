import React from 'react';
import { render } from '../../utils/testUtils';
import { CourseContent } from './CourseContent';


describe('components/CourseContent', () => {
  it('should render', () => {
    const { container } = render(<CourseContent />);
    expect(container).toMatchSnapshot();
  });
});
