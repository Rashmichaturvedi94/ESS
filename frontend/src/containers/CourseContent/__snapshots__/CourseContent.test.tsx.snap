// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CourseContent should render 1`] = `
<div>
  <div
    class="sc-bdnxRM iNeoZH"
  >
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <div
        class="MuiBox-root MuiBox-root-2"
        style="margin: 2px;"
      >
        <div
          class="MuiBox-root MuiBox-root-3"
          style="padding: 2px;"
        >
          <h2
            class="MuiTypography-root MuiTypography-h3 MuiTypography-alignLeft"
          />
        </div>
        <div
          style="width: 640px; height: 360px;"
        />
      </div>
      <hr
        class="MuiDivider-root MuiDivider-vertical"
        style="color: black; width: 5px;"
      />
      <div
        class="MuiBox-root MuiBox-root-4"
      >
        <ul
          class="MuiGridList-root"
          style="margin: -2px;"
        >
          <p
            class="MuiTypography-root sc-dlnjwi FzyQW MuiTypography-body1"
            style="width: 50%; height: 184px; padding: 2px;"
          >
            Lessons
          </p>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
